src/main/java/gov/nasa/cumulus/metadata/aggregator/MetadataFilesToEcho.java

```diff
@@ -557,6 +559,81 @@ public IsoGranule readIsoMendsMetadataFile(String s3Location, Document doc, XPat
+				setGranuleBoundingBox(
+						Double.parseDouble(xpath.evaluate(IsoMendsXPath.NORTH_BOUNDING_COORDINATE, doc)),
+						Double.parseDouble(xpath.evaluate(IsoMendsXPath.SOUTH_BOUNDING_COORDINATE, doc)),
+						Double.parseDouble(xpath.evaluate(IsoMendsXPath.EAST_BOUNDING_COORDINATE, doc)),
+						Double.parseDouble(xpath.evaluate(IsoMendsXPath.WEST_BOUNDING_COORDINATE, doc)));
+			}
+		}
+		/**
+		 * There shall be no more logic of "if there is orbit, then there shall be no footprint" anymore
+		 * this function will solely depends on cumulus collection config to determine the collection is
+		 * GEODETIC, CARTICIAN , ORBIT
+		 * if isoXMLSpatialTypeEnumHashSet contains FOOTPRINT, means, the collection is either GEODETIC or CARTICIAN
+		 */
+		if(this.isoXMLSpatialTypeEnumHashSet.contains(MENDsIsoXMLSpatialTypeEnum.FOOTPRINT)) {
+			AdapterLogger.LogDebug(this.className + " Based on MENDsIsoXMLSpatialTypeEnum, processing FOOTPRINT");
+			try {
+				((IsoGranule) granule).setPolygon(xpath.evaluate(IsoMendsXPath.POLYGON, doc));
+				String line = xpath.evaluate(IsoMendsXPath.LINE, doc);
+				if (line != null && !line.isEmpty()) {
+					granule.getGranuleCharacterSet().add(createGranuleCharacter(line,"line"));
+				}
+			} catch (XPathExpressionException e) {
+				// Ignore if unable to parse for footprint since it isn't required for ingest
+				AdapterLogger.LogWarning(this.className + " Not able to extract MENDS footprint: " + e);
+			}
+		}
+		if(this.isoXMLSpatialTypeEnumHashSet.contains(MENDsIsoXMLSpatialTypeEnum.ORBIT)) {
+			AdapterLogger.LogDebug(this.className + " Based on MENDsIsoXMLSpatialTypeEnum, processing ORBIT");
+			((IsoGranule) granule).setOrbit(xpath.evaluate(IsoMendsXPath.ORBIT, doc));
+		}
+		return ((IsoGranule) granule);
+	}
+
+	public boolean isOrbitExisting(String orbitStr) {
+		if(StringUtils.isEmpty(StringUtils.trim(orbitStr))) {
+			return false;
+		} else{
+			try {
+				Pattern p = Pattern.compile("AscendingCrossing:\\s?(.*)\\s?StartLatitude:\\s?(.*)\\s?StartDirection:\\s?(.*)\\s?EndLatitude:\\s?(.*)\\s?EndDirection:\\s?(.*)");
+				Matcher m = p.matcher(orbitStr);
+				boolean foundOrbitalData = false;
+				foundOrbitalData = m.find();
+				String ascendingCrossingStr = StringUtils.trim(m.group(1));
+				String startLatitudeStr = StringUtils.trim(m.group(2));
+				String startDirectionStr = StringUtils.trim(m.group(3));
+				String endLatitudeStr = StringUtils.trim(m.group(4));
+				String endDirectionStr = StringUtils.trim(m.group(5));
+
+				/** to verify the Orbit string using as-tight-as-possible logic to make sure the orbitStr is parsable
+				 *  and not anyone of the item is "None"
+				 */
+				if (foundOrbitalData && BoundingTools.allParsable(ascendingCrossingStr, startLatitudeStr, endLatitudeStr) && (
+						!StringUtils.equalsIgnoreCase(ascendingCrossingStr, "None") &&
+								!StringUtils.equalsIgnoreCase(startLatitudeStr, "None") &&
```

Comment:
Why are we comparing this against `None`?

Reply:
the isOrbitExisting function and its unit tests are removed.  It was needed while we were programming SWOT logic "if no orbit existed, then add polygon".  Now we are depending on the isoXMLSpatialType